etl/steps/data/garden/faostat/2023-02-22/additional_variables.py

```diff
@@ -0,0 +1,707 @@
+    assert set(spared_land["unit"]) == set(["tonnes per hectare", "tonnes"]), error
+
+    # Transpose dataframe.
+    spared_land = spared_land.pivot(
+        index=["country", "year", "item"], columns=["element"], values="value"
+    ).reset_index()
+
+    # Fix column name after pivotting.
+    spared_land.columns = ["country", "year", "item", "Yield", "Production"]
+
+    # Add columns for production and yield for a given reference year.
+    reference_values = spared_land[spared_land["year"] == REFERENCE_YEAR].drop(columns=["year"])
+    spared_land = pd.merge(
+        spared_land, reference_values, on=["country", "item"], how="left", suffixes=("", f" in {REFERENCE_YEAR}")
+    )
+
+    # Drop countries for which we did not have data in the reference year.
+    spared_land = spared_land.dropna().reset_index(drop=True)
+
+    # Calculate area harvested that would be required given current production, but with the yield of the reference year.
+    spared_land[f"Area with yield of {REFERENCE_YEAR}"] = (
+        spared_land["Production"] / spared_land[f"Yield in {REFERENCE_YEAR}"]
+    )
+    # Calculate the real area harvested (given the current production and yield).
+    spared_land["Area"] = spared_land["Production"] / spared_land["Yield"]
+
+    # Keep only required columns
+    spared_land = spared_land[["country", "year", "item", "Area", f"Area with yield of {REFERENCE_YEAR}"]].reset_index(
+        drop=True
+    )
+
+    # Add total area for all crops.
+    all_crops = (
+        spared_land.groupby(["country", "year"], as_index=False, observed=True)
+        .agg({"Area": sum, f"Area with yield of {REFERENCE_YEAR}": sum})
+        .assign(**{"item": "All crops"})
+    )
+    spared_land = pd.concat([spared_land, all_crops], ignore_index=True)
+
+    # Calculate the spared land in absolute value, and as a percentage of the land we would have used with no yield increase.
+    spared_land["Spared land"] = spared_land[f"Area with yield of {REFERENCE_YEAR}"] - spared_land["Area"]
+    spared_land["Spared land (%)"] = (
+        100 * spared_land["Spared land"] / spared_land[f"Area with yield of {REFERENCE_YEAR}"]
+    )
+
+    # Create a table with the necessary columns, set an appropriate index, and sort conveniently.
+    tb_spared_land = Table(
+        spared_land.set_index(["country", "year", "item"], verify_integrity=True).sort_index(),
+        short_name="land_spared_by_increased_crop_yields",
+        underscore=True,
+    )
+
+    return tb_spared_land
+
+
+def generate_food_available_for_consumption(df_fbsc: pd.DataFrame) -> Table:
+    # Element code for "Food available for consumption" of faostat_fbsc (in kilocalories per day per capita).
+    ELEMENT_CODE_FOR_PER_CAPITA_FOOD = "0664pc"
+    # Expected unit.
+    CONSUMPTION_UNIT = "kilocalories per day per capita"
+
+    # df_fbsc[df_fbsc["unit"].str.contains("kilocal")][["element_code", "element", "unit"]].drop_duplicates()
```

Comment:
You could maybe delete this comment if you are not using it anymore

Reply:
Ups, true!